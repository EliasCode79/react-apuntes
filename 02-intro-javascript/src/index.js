//  Operador ternario

let msj = 'Este es un msj';

let esVerdad = true

let msjSend = ( esVerdad && es)