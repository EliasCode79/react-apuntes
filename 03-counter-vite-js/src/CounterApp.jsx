import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const handler = (event) => {
		// console.log(event)
		console.log('+1');
		value = 100;
	};

	return (
		<>
			<h1>Counter App</h1>
			<h2>{value}</h2>

			{/* onclick envia por defecto un event, en react podemos llamar a una funcion si (), cuando se envia un parametro */}
			<button onClick={handler}>+1</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};

CounterApp.defaultPros = {
	value: 'No hay numero',
};
