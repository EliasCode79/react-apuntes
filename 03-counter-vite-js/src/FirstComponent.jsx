export const FirtsApp = () => {
	return <h2>Elias Franco</h2>;
};
