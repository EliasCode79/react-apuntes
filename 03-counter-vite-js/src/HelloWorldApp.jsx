// no estamos importando React, porque ya los estamos importando de manera global con Vite
export const App = () => {
	return <h1>Hola mundo</h1>;
};
