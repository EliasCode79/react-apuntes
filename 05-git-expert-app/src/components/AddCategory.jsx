import { useState } from 'react';

export const AddCategory = () => {
	const [inputValue, setInputValue] = useState('init state');

	// este metodod es para la entrada del input
	const onInputChange = ({ target }) => {
		// console.log(target.value);
		setInputValue(target.value);
	};

	// este metodo evita q el q el formulario se envie.
	const onSubmit = (event) => {
		event.preventDefault();
		console.log(inputValue);
	};

	return (
		<form onSubmit={(event) => onSubmit(event)}>
			<input type='text' placeholder='introduce un texto' value={inputValue} onChange={onInputChange} />;
		</form>
	);
};
