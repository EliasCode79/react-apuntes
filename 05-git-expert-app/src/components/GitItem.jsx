import PropTypes from 'prop-types';

export const GitItem = ({ url, title }) => {
	return (
		<div className='card'>
			<img className='card' src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GitItem.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};
