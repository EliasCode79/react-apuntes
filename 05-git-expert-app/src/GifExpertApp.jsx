import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['one punch', 'dragon ball']);

	const onAddCategory = () => {
		let nueva = 'demon slayer';
		setCategories([...categories, nueva]);
	};

	return (
		<>
			{/* title */}
			<h1>GifExpertApp</h1>

			{/* input */}
			<AddCategory setCategories={setCategories} />
			{/* grid table */}

			{/* <button onClick={(e) => onAddCategory()}>Agregar</button> */}
			<ol>
				{/* Para recorrer un array usamo el map, a los elementos li tenemos q agregarles un key para q no nos presente problemas en la consola */}
				{categories.map((categorie) => {
					return <li key={categorie}>{categorie}</li>;
				})}
			</ol>

			{/* grid item */}
		</>
	);
};
