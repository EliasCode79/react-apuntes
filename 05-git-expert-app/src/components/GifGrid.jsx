import { GitItem } from './GitItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	const { images, isLoading } = useFetchGifs(category);

	return (
		<>
			<h3>{category}</h3>

			{/* Esta expresion si se cumplen las 2 retorna lo q es la segunda condicion */}
			{isLoading && <h2>Cargando....</h2>}

			<div className='card-grid'>
				{images.map((image) => (
					<GitItem key={image.id} {...image} />
				))}
			</div>
		</>
	);
};
