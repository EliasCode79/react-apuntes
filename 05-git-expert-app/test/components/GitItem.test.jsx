import { render, screen } from '@testing-library/react';
import { GitItem } from '../../src/components';

describe('test de <GitItem />', () => {
	const title = 'saitama';
	const url = 'https://google.com/';

	test('Snapshot del componente', () => {
		const { container } = render(<GitItem url={url} title={title} />);
		expect(container).toMatchSnapshot();
	});

	test('ver q los titulos la url y el title sean iguales', () => {
		render(<GitItem url={url} title={title} />);
		// expect(screen.getByRole('img').src).toBe(url);

		// esto es lo mismo q arriba pero aca lo hicimos mas legible.
		const { src, alt } = screen.getByRole('img');
		expect(alt).toBe(title);
		expect(src).toBe(url);
	});

	test('Debe mostrar el titulo del componente', () => {
		render(<GitItem url={url} title={title} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
