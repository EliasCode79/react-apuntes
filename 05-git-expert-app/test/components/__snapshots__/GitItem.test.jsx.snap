// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test de <GitItem /> Snapshot del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      class="card"
      src="https://google.com/"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
